cmd/consensus/main.go

```diff
@@ -616,10 +611,12 @@ func main() {
 				mainMetrics,
 			)
 
+			telemetryConsumer := notifications.NewTelemetryConsumer(logger, notifier)
+
+			// TODO(leo): move these to NewTelemetryConsumer
 			notifier.AddParticipantConsumer(telemetryConsumer)
 			notifier.AddCommunicatorConsumer(telemetryConsumer)
-			notifier.AddFinalizationConsumer(telemetryConsumer)
```

Comment:
The `AddFinalizationConsumer` is removed, because it's redundant. We could just call the following instead:
```
notifier.AddOnBlockFinalizedConsumer(telemetryConsumer.OnBlockFinalized)
notifier.AddOnBlockIncorporatedConsumer(telemetryConsumer.OnBlockIncorporated)
```

But we also don't want to do the above, instead, we passed the notifier to the constructor. See line 614:

```
telemetryConsumer := notifications.NewTelemetryConsumer(logger, notifier)
```